modules/wr_fronius/fronius.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+from datetime import datetime, timezone
+import os
+import requests
+import sys
+import traceback
+
+wrfroniusip = str(sys.argv[1])
+wrfronius2ip = str(sys.argv[2])
+wrfroniusisgen24 = str(sys.argv[3])
+
+Debug = int(os.environ.get('debug'))
+myPid = str(os.getpid())
+
+
+def DebugLog(level, message):
+    if Debug >= level:
+        local_time = datetime.now(timezone.utc).astimezone()
+        print(local_time.strftime(format="%Y-%m-%d %H:%M:%S") + ": PID: " + myPid + ": " + message)
+
+
+DebugLog(2, 'Fronius IP: ' + wrfroniusip)
+DebugLog(2, 'Fronius 2 IP: ' + wrfronius2ip)
+DebugLog(2, 'Fronius Gen24: ' + wrfroniusisgen24)
+
+params = (
+    ('Scope', 'System'),
+)
+regex='^-?[0-9]+$'
+
+# Auslesen eines Fronius Symo WR über die integrierte API des WR.
+# Rückgabewert ist die aktuelle Wirkleistung in [W].
+response = requests.get('http://'+wrfroniusip+'/solar_api/v1/GetPowerFlowRealtimeData.fcgi', params=params, timeout=3)
+pvwatttmp = response.json()
+DebugLog(1, 'response: ' + str(response))
+DebugLog(2, 'response_pvwatt: ' + str(pvwatttmp))
+try:
+    pvwatt = int(pvwatttmp["Body"]["Data"]["Site"]["P_PV"])
+except:
+    # Wenn WR aus bzw. im Standby (keine Antwort), ersetze leeren Wert durch eine 0
+    pvwatt = 0
```

Comment:
Wenn der WR nicht antwortet (weil kein Strom oder kein Netzwerk), dann wird es nicht zu dieser Zeile hier kommen. In dem Fall fliegt schon in Zeile 33 eine Exception und das Script bricht ab. Das Verhalten finde ich OK. Es kann durchaus sein, dass nur gerade jemand den Netzwerkswitch neu startet und dann gibt es eben kurz keine neuen Daten und dann wäre es kontraproduktiv deswegen die Leistung auf 0 zu setzen.

Wenn es tatsächlich unbedingt auf 0 gesetzt werden soll, muss das Try/except aber in Zeile 33.

Wenn der WR im Standby ist, dann ist der Kommentar zutreffend: P_PV ist nachts `null` und nicht wie man erwarten könnte `0`. Das könnte noch eine hilfreiche Info in dem Kommentar sein.

Noch eindeutiger fände ich aber sowas:
```python
pvwatt_float = pvwatttmp["Body"]["Data"]["Site"]["P_PV"]
pvwatt = 0 if pvwatt_float is None else int(pvwatt_float)
```

Reply:
Hier sollte nur P_PV: null behandelt werden. An dieser Stelle ist doch gesichert, dass der WR lief und mit einem JSON-Response geantwortet hat.
Da der WR ansonsten auch den Hausverbrauch liefert, macht ein Ausschalten wenig Sinn, da sonst die restliche Steuerlogik in der OpenWB nicht mehr funktionieren kann. Deshalb denke ich auch, dass es keine gesonderte Fehlerbehandlung für Zeilen 32, 33 braucht. 
Für das eigentliche Problem hier, bin ich über einen eleganten Einzeiler gestolpert:
```python
# Ohne PV Produktion liefert der WR 'null', ersetze durch Zahl 0
pvwatt = int(pvwatttmp["Body"]["Data"]["Site"]["P_PV"] or 0)

Reply:
Ja, stimmt, mir dem `or 0` kannst du es auch lösen.